### PR TITLE
Add tests for literal types

### PR DESCRIPTION
**Pull request by pshao25**
# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/payload/content-negotiation/main.tsp [hunk 1]
```diff
@@ -54,7 +54,7 @@ namespace DifferentBody {
 
   model PngImageAsJson {
     @header contentType: "image/jpeg";
-    content: bytes;
+    content: string;
   }
```
Comment (pshao25) [inline packages/cadl-ranch-specs/http/payload/content-negotiation/main.tsp:58]: This is an expected fix

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/payload/content-negotiation/main.tsp:58]: @timotheeguerin 

Should it be
```ts
  model PngImageAsJson {
    @header contentType: "application/json";
    content: bytes;
  }
```
(or without `contentType`)

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/payload/content-negotiation/main.tsp:58]: e.g. https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/ContentSafety/models.tsp#L102-L104

File: packages/cadl-ranch-specs/http/type/property/optionality/main.tsp [hunk 2]
```diff
@@ -128,6 +128,55 @@ model CollectionsModelProperty is ModelTemplate<StringProperty[]>;
 interface CollectionsModel
   extends OperationsTemplate<CollectionsModelProperty, "[{'property': 'hello'}, {'property': 'world'}]"> {}
 
+// Model with optional string literal property
+@doc("Model with string literal property")
+model StringLiteralProperty is ModelTemplate<"hello">;
+@route("/string/literal")
+@operationGroup
+interface StringLiteral extends OperationsTemplate<StringLiteralProperty, "hello"> {}
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/property/optionality/main.tsp:136]:
```suggestion
interface StringLiteral extends OperationsTemplate<StringLiteralProperty, '"hello"'> {}
```

quote on the string?

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 3]
```diff
@@ -3723,6 +4063,86 @@ Expected input body:
...
 ```
 
+### Type_Property_ValueTypes_StringLiteral_get
+
+- Endpoint: `get /type/property/value-types/string/literal`
+
+Expected response body:
+
+```json
+{"property": hello}
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:4073]: see https://github.com/Azure/cadl-ranch/pull/426/files#r1335605252

File: packages/cadl-ranch-specs/http/type/property/optionality/main.tsp (hunk 2 above)
Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/type/property/optionality/main.tsp:136]: can't do single quotes in typespec but you can still escape with `\`

**Review by tadelesh — APPROVED** [review]
nit: string quote

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 4]
```diff
@@ -4019,6 +4019,46 @@ Expected request body:
...
 ```
 
+### Type_Property_Optional_BooleanLiteral_getAll
+
+- Endpoint: `get /type/property/optional/boolean/literal/all`
+
+Expected response body:
+
+```json
+{"property": doc}
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:4029]: it's still there. please fix all of these.

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 5]
```diff
@@ -4019,6 +4019,46 @@ Expected request body:
...
+
+### Type_Property_Optional_BooleanLiteral_putDefault
+
+- Endpoint: `put /type/property/optional/boolean/literal/default`
+
+Expected request body:
+
+```json
+{}
+```
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:4060]: why do we expect empty body in Type_Property_Optional_BooleanLiteral_putDefault?

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/cadl-ranch-summary.md:4060]: ok, got it, it's just `?`

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 6]
```diff
@@ -4019,14 +4019,54 @@ Expected request body:
...
+
 ### Type_Property_Optional_Bytes_getAll
 
 - Endpoint: `get /type/property/optional/bytes/all`
 
 Expected response body:
 
 ```json
-{"property": doc}
+{"property": aGVsbG8sIHdvcmxkIQ==}
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:4069]: this not correct? should be `"aGVsbG8sIHdvcmxkIQ=="`

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 7]
```diff
@@ -4146,7 +4186,7 @@ Expected request body:
 Expected response body:
 
 ```json
-{"property": doc}
+{"property": 2022-08-26T18:38:00Z}
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:4189]: same here

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 8]
```diff
@@ -4186,7 +4226,7 @@ Expected request body:
 Expected response body:
 
 ```json
-{"property": doc}
+{"property": P123DT22H14M12.011S}
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:4229]: here

**Review by iscai-msft — APPROVED** [review]

**Pull request merged**